### PR TITLE
Story2

### PR DESCRIPTION
**Pull request by OksanaBakhaieva**

### CODE REVIEW

File: app/Http/Controllers/RoomAPIController.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+use App\Models\Room;
+use Illuminate\Http\Request;
+
+class RoomAPIController extends Controller
+{
+   public function find(int $id): \Illuminate\Http\JsonResponse
+   {
+       $room = Room::find($id);
+       if($room){
```
Comment (ashleycoles) [inline app/Http/Controllers/RoomAPIController.php]: You should instead check to see if there is not a room here, and then return your 404 response.

It's always good practice to 'fail early', meaning you should check for failures (like the room not existing) as soon as possible.

This way, you can have something like

```
if (!$room) {
   // return the 404
}

// Return the success response
```

Eliminating the need for the else which does help to make the code simpler

File: app/Http/Controllers/RoomAPIController.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\Room;
+use Illuminate\Http\Request;
+
+class RoomAPIController extends Controller
+{
+   public function find(int $id): \Illuminate\Http\JsonResponse
+   {
+       $room = Room::find($id);
```
Comment (ashleycoles) [inline app/Http/Controllers/RoomAPIController.php]: Here you should really be using with() to load the type straight away - like what was done in story 1

File: app/Http/Controllers/RoomAPIController.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                   'name' => $room->name,
+                   'rate' => $room->rate,
+                   'image' => $room->image,
+                   'min_capacity' => $room->min_capacity,
+                   'max_capacity' => $room->max_capacity,
+                   'description' => $room->description,
+                   'type' => [
+                       'id' => $room->type->id,
+                       'name' => $room->type->name,
+                    ]
```
Comment (ashleycoles) [inline app/Http/Controllers/RoomAPIController.php]: You don't need to do all this stuff, once you add a with to the query you should be able to just put $room in data

File: app/Models/Room.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+class Room extends Model
+{
+    use HasFactory;
+    public function type(): BelongsTo
```
Comment (ashleycoles) [inline app/Models/Room.php]: Add empty line above this method

File: app/Models/Type.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class Type extends Model
+{
+    use HasFactory;
+    public function rooms(): HasMany
```
Comment (ashleycoles) [inline app/Models/Type.php]: Add empty line above this method

File: database/factories/RoomFactory.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    public function definition(): array
+    {
+        return [
+            'name' => $this->faker->sentence(2),
+            'description' => $this->faker->paragraph(2),
+            'image' => $this->faker->imageUrl($width = 400, $height = 400),
+            'min_capacity' => rand(1,2),
+            'max_capacity' => rand(2,6),
+            'rate'=>rand(50,300),
+            'type_id'=>Type::factory()
```
Comment (ashleycoles) [inline database/factories/RoomFactory.php]: Spaces around the => please

**Review by ashleycoles — CHANGES_REQUESTED** [review]

**Comment by Anerkiki** [comment]
We were a little unsure with RoomAPIController.php, but think this is alright?

**Review by ashleycoles — APPROVED** [review]

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**